Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java

```diff
@@ -426,9 +439,16 @@ protected String generateId(String prefix) {
     }
 
     @Override
-    public void deleteConcept(String id) throws LuceneException, IndexerRunningException {
+    public void deleteConcept(String id, String userName) throws LuceneException, IndexerRunningException {
         ConceptEntry concept = getConceptEntry(id);
         concept.setDeleted(true);
+ChangeEvent changeEvent = new ChangeEvent();
+        changeEvent.setType(ChangeEventConstants.DELETION);
+        changeEvent.setDate(new Date());
+        changeEvent.setUserName(userName);
+        List<ChangeEvent> changeEventList = new ArrayList<ChangeEvent>();
+        changeEventList.addAll(concept.getChangeEvents());
+        changeEventList.add(changeEvent);
```

Comment:
Why do you create a new list here? As before, your concept entry should provide a method to add a new change event.